backtrace-library/src/main/java/backtraceio/library/common/BatteryStateHelper.java

```diff
@@ -0,0 +1,244 @@
+import backtraceio.library.logger.BacktraceLogger;
+
+public class BatteryStateHelper extends BroadcastReceiver implements LifecycleObserver {
+    private static Context context;
+    private static BacktraceDatabase database;
+    private static boolean enabled;
+    private static String batteryHealth, batteryLevel, batteryChargingSource, batteryChargingStatus, batteryTechnology, batteryTemperature;
+    private static IntentFilter intentFilter;
+    private static BatteryStateHelper batteryStateHelper;
+    private static String TAG = "BatteryStateHelper";
+
+    private BatteryStateHelper(Context context, Database database) {
+        if (BatteryStateHelper.context == null){
+            BatteryStateHelper.context = context;
+            if (database != null) {
+                BatteryStateHelper.database = (BacktraceDatabase)database;
+            }
+        }
+        if (intentFilter == null) {
+            intentFilter = new IntentFilter();
+            intentFilter.addAction(Intent.ACTION_BATTERY_CHANGED);
+        }
+    }
+
+    public static BatteryStateHelper getInstance(Context context, Database database) {
+        if (batteryStateHelper == null) {
+            if (context == null)
+                return null;
+            batteryStateHelper = new BatteryStateHelper(context, database);
+        }
+        return  batteryStateHelper;
+    }
+
+    public static void setDatabase(Database database) {
+        batteryStateHelper.database = (BacktraceDatabase) null;
+    }
+
+    public static HashMap<String, String> getValues() {
+        HashMap<String, String> result = new HashMap<String, String>();
+        if (!enabled) {
+            result.put("battery.health", "N/A");
+            result.put("battery.level", "N/A");
+            result.put("battery.charging.source", "N/A");
+            result.put("battery.charging.status", "N/A");
+            result.put("battery.technology", "N/A");
+            result.put("battery.temperature", "N/A");
+            return result;
+        }
+        result.put("battery.health", batteryHealth);
+        result.put("battery.level", batteryLevel);
+        result.put("battery.charging.source", batteryChargingSource);
+        result.put("battery.charging.status", batteryChargingStatus);
+        result.put("battery.technology", batteryTechnology);
+        result.put("battery.temperature", batteryTemperature);
+        return result;
+    }
+
+    public void disable() {
+        enabled = false;
+        if (getInstance(null, null) != null) {
+            context.unregisterReceiver(batteryStateHelper);
+            ProcessLifecycleOwner.get().getLifecycle().removeObserver(batteryStateHelper);
+        }
+    }
+
+    public void enable() {
+        if (getInstance(null, null) != null) {
+            enabled = true;
+            ProcessLifecycleOwner.get().getLifecycle().addObserver(batteryStateHelper);
+            Intent batteryStatus = context.registerReceiver(null, intentFilter);

```

Comment:
Do we need to make two `context.registerReceiver` calls in this function? Can we remove this line?